pkg/collector/corechecks/system/iostats_pdh_windows.go

```diff
@@ -117,6 +117,11 @@ func (c *IOCheck) Run() error {
 			tagbuff.WriteString("device:")
 			tagbuff.WriteString(inst)
 			tags := []string{tagbuff.String()}
+
+			if matched, _ := regexp.MatchString(`^[A-z]:`, inst); !matched {
```

Comment:
I think it makes more sense to compile this and move it to the package scope like it is already done for the other regexes in the file:
https://github.com/DataDog/datadog-agent/blob/7ddd10fe3a6676315b731b9033fe4e27e875ab4e/pkg/collector/corechecks/system/iostats_pdh_windows.go#L29-L30